rpc/src/p2p.rs

```diff
@@ -1,17 +1,61 @@
+use celestia_types::p2p::{
+    AddrInfo, BandwidthStats, Connectedness, PeerId, Reachability, ResourceManagerStats,
+};
 use jsonrpsee::proc_macros::rpc;
-use serde::{Deserialize, Serialize};
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct AddrInfo {
-    #[serde(rename = "ID")]
-    pub id: String,
-    // TODO: multiaddr
-    #[serde(rename = "Addrs")]
-    pub addrs: Vec<String>,
-}
 
 #[rpc(client)]
 pub trait P2P {
+    #[method(name = "p2p.BandwidthForPeer")]
+    async fn p2p_bandwidth_for_peer(&self, peer_id: &PeerId) -> Result<BandwidthStats, Error>;
+
+    #[method(name = "p2p.BandwidthForProtocol")]
+    async fn p2p_bandwidth_for_protocol(&self, protocol_id: &str) -> Result<BandwidthStats, Error>;
+
+    #[method(name = "p2p.BandwidthStats")]
+    async fn p2p_bandwidth_stats(&self) -> Result<BandwidthStats, Error>;
+
+    #[method(name = "p2p.BlockPeer")]
+    async fn p2p_block_peer(&self, peer_id: &PeerId);
+
+    #[method(name = "p2p.ClosePeer")]
+    async fn p2p_close_peer(&self, peer_id: &PeerId);
+
+    #[method(name = "p2p.Connect")]
+    async fn p2p_connect(&self, address: &AddrInfo);
```

Comment:
Please add docs to those with a warning that errors are not reported and why, with the link to the relevant issue

Reply:
actually could you try an `Option<Error>` there for fun? :smiling_imp:

Reply:
Created https://github.com/eigerco/celestia-node-rs/issues/53 for tracking and added a note to each affected method.

As for the option - I've tried that, but `jsonprsee` wants either `Result` or `RpcResult` there:
```
error: The return type must be Result or RpcResult
  --> rpc/src/p2p.rs:19:57
   |
19 |     async fn p2p_block_peer(&self, peer_id: &PeerId) -> Option<()>;

```